torch/nn/functional.py

```diff
@@ -3953,3 +3953,188 @@ def multi_head_attention_forward(query,                           # type: Tensor
         return attn_output, attn_output_weights.sum(dim=1) / num_heads
     else:
         return attn_output, None
+
+
+def multi_head_attention_in_projection(seq, num_heads, in_proj_weight, in_proj_bias=None):
```

Comment:
Have we verified that calling this module thrice is as efficient as calling it with a single concatenated weight generating q,k and v?

Reply:
We have not. That is something that @zhangguanheng66 offered to do. It's not guaranteed that this module is even going to persist in this form at the moment, so it's a premature to do so.